src/fides/api/ops/service/connectors/saas_query_config.py

```diff
@@ -282,8 +294,15 @@ def generate_update_param_values(  # pylint: disable=R0914
         param_values: Dict[str, Any] = {}
         for param_value in saas_request.param_values or []:
             if param_value.references:
+                reference: FidesopsDatasetReference = (
+                    FidesopsDatasetReference.parse_obj(
+                        pydash.get(self.secrets, param_value.references[0])
+                    )
+                    if isinstance(param_value.references[0], str)
+                    else param_value.references[0]
+                )
                 param_values[param_value.name] = pydash.get(
-                    collection_values, param_value.references[0].field
+                    collection_values, reference.field
```

Comment:
in reality, this is not going to work - we cannot resolve any references outside of the given collection for updates at this point (even if they are within the same dataset). we won't get those values passed in on our input `row`. but i decided to update this code anyway to keep things "consistent". perhaps it needs a comment for clarification? also open to other ideas...

Reply:
yeah, that's right. we're limited here (in the erasure workflow) to references on the current collection. i can definitely include a comment to clarify some of this